### PR TITLE
Add uqs' keymaps for various boards and ploopy mouse

### PR DESCRIPTION
**Pull request by uqs**
This contains a log of my 1 year journey to switch to ColemakDH. Some
other features are:
- nice Alt+Tab using the layer hold as Alt hold
- OSM that work (for me)
- ugly but significantly less verbose way to set combos
- some _Static_asserts to stop me from being silly

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/handwired/dactyl_manuform/5x6/keymaps/uqs/config.h [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#pragma once
```
Comment (drashna) [inline keyboards/handwired/dactyl_manuform/5x6/keymaps/uqs/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: keyboards/ploopyco/mouse/keymaps/uqs/keymap.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+enum custom_keycodes {
+    DBL_CLK_NO = SAFE_RANGE,
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case DBL_CLK_NO:
+            if (record->event.pressed) {
+                //SEND_STRING(SS_TAP(X_BTN2) SS_DELAY(300) SS_TAP(X_BTN2) SS_DELAY(300) SS_TAP(X_N));
+                tap_code(KC_BTN2);
```
Comment (drashna) [inline keyboards/ploopyco/mouse/keymaps/uqs/keymap.c:27]: Just a heads up, you may want to look at the deferred executor stuff.  This allows for adding waits, but without hard locks, like this. 

It's a bit more complicated, but it is very rewarding.

File: users/uqs/config.h [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#pragma once
```
Comment (drashna) [inline users/uqs/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

File: users/uqs/config.h [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+#define DYNAMIC_KEYMAP_LAYER_COUNT 6  // default is 4 for VIA builds
+
+#define TAPPING_TOGGLE 2  // number of taps for a toggle-on-tap
+#define TAPPING_TERM 170  // ms to trigger tap
+// https://precondition.github.io/home-row-mods
+#define TAPPING_FORCE_HOLD  // make tap-then-hold _not_ do key auto repeat
+#define IGNORE_MOD_TAP_INTERRUPT
+#define PERMISSIVE_HOLD  // I don't think this works for me, hence I rolled my own implementation.
+//#define PREVENT_STUCK_MODIFIERS
```
Comment (drashna) [inline users/uqs/config.h:21]:
```suggestion
```

File: users/uqs/config.h (hunk 3 above)
Comment (drashna) [reply users/uqs/config.h:1]: Also, this file could use a formatting pass.

File: users/uqs/rules.mk [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+MOUSEKEY_ENABLE = yes
+
+# Disable all the unused stuff.
+SPACE_CADET_ENABLE = no
+COMMAND_ENABLE = no
+MAGIC_ENABLE = no
+
+# From https://michael.stapelberg.ch/posts/2021-05-08-keyboard-input-latency-qmk-kinesis/
+# adds about 900 bytes!
+DEBOUNCE_TYPE = asym_eager_defer_pk
```
Comment (drashna) [inline users/uqs/rules.mk:20]:
```suggestion
    SRC += uqs.c

    RGBLIGHT_ENABLE ?= yes      # Enable keyboard RGB underglow
    LTO_ENABLE ?= yes  # disables the legacy TMK Macros and Functions features

    UCIS_ENABLE = yes
    LEADER_ENABLE = yes
    COMBO_ENABLE = yes
    MOUSEKEY_ENABLE = yes

    # Disable all the unused stuff.
    SPACE_CADET_ENABLE = no
    COMMAND_ENABLE = no
    MAGIC_ENABLE = no

    # From https://michael.stapelberg.ch/posts/2021-05-08-keyboard-input-latency-qmk-kinesis/
    # adds about 900 bytes!
    DEBOUNCE_TYPE = asym_eager_defer_pk
```

Also, I would NOT recommend the asym eager defer pk algo for non-arm controllers, IIRC.

File: users/uqs/uqs.h [hunk 6]
```diff
@@ -0,0 +1,75 @@
+// vi:et sw=4:
```
Comment (drashna) [inline users/uqs/uqs.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15n

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/handwired/dactyl_manuform/5x6/keymaps/uqs/config.h (hunk 1 above)
Comment (uqs) [reply keyboards/handwired/dactyl_manuform/5x6/keymaps/uqs/config.h:1]: Sure, I added the license blurb to all the .c files, as the .h are so short and pretty generic anyway. If that 2-line thing is enough, I'll add that later.

File: keyboards/ploopyco/mouse/keymaps/uqs/keymap.c (hunk 2 above)
Comment (uqs) [reply keyboards/ploopyco/mouse/keymaps/uqs/keymap.c:27]: Thanks, I'll add a TODO, but i'm currently not really using that mouse and it somehow didn't work in-game anyway (maybe too quick still?)

File: users/uqs/rules.mk (hunk 5 above)
Comment (uqs) [reply users/uqs/rules.mk:20]: care to explain? is it too slow on the AVRs? which ones are suitable and faster for AVR?

Comment (drashna) [reply users/uqs/rules.mk:20]: I'd have to dig up the conversation, but yeah, it's slower and more computationally expensive on AVR.

Comment (uqs) [reply users/uqs/rules.mk:20]: hmpf, how to make this conditional in my users/uqs/rules.mk though? It seems to get read before $(MCU) was set :/

Comment (drashna) [reply users/uqs/rules.mk]: post_rules.mk may work, but I'm not sure.

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**